lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -384,6 +402,54 @@ TfTokenVector getProxyShapePurposes(const Ufe::Path& path)
     return purposes;
 }
 
+bool isAttributeEditAllowed(const PXR_NS::UsdAttribute& attr, std::string* errMsg)
+{
+    // get the property spec in the edit target's layer
+    const auto& prim = attr.GetPrim();
+    const auto& stage = prim.GetStage();
+    const auto& editTarget = stage->GetEditTarget();
+
+    // get the index to edit target layer
+    const auto targetLayerIndex = findLayerIndex(attr, editTarget.GetLayer());
+
+    // get the strength-ordered ( strong-to-weak order ) list of property specs that provide
+    // opinions for this property.
+    const auto& propertyStack = attr.GetPropertyStack();
+
+    // HS March 15th,2021
```

Comment:
As I mentioned I think the loop is an over-elaboration.  What you want is to compare the layer at the top of the property stack to the target layer index, something like this:

SdfLayerHandle topLayer = propertyStack.front().GetLayer();
if (findLayerIndex(attr, topLayer) < targetLayerIndex) {
    std::string err = TfStringPrintf( // etc.

Reply:
I replaced the loop if it makes logics easier to follow.

addressed ac2bb4dc732e9862957b25a17eb9f6f7719adcf3

Reply:
Thanks, makes more sense to me at least :)